src/mds/MDSRank.cc

```diff
@@ -3852,3 +3880,32 @@ void MDSRank::send_task_status() {
 
   schedule_update_timer_task();
 }
+
+void MDSRank::schedule_inmemory_logger() {
+  dout(20) << __func__ << dendl;
+  timer.add_event_after(extraordinary_events_dump_interval,
+                        new LambdaContext([this]() {
+                          inmemory_logger();
+                        }));
+}
+
+void MDSRank::inmemory_logger() {
+  if (client_eviction_dump ||
+      beacon.missed_beacon_ack_dump ||
+      beacon.missed_internal_heartbeat_dump) {
+    //dump the in-memory logs if any of these events occured recently
+    reset_event_flags();
+    g_ceph_context->_log->dump_recent();
+  }
+
+  //reschedule if it's still enabled
+  if (extraordinary_events_dump_interval && log_level < 10) {
```

Comment:
@lxbsz 
I think we should avoid checking `log_level < 10` here. Because to implement this check correctly, it needs reading the config again. May be reading `gather_level` again too. Which would be expensive in the `tick()`. Right?

Reply:
No need to check the `log_level` here, because you have disable the `extraordinary_events_dump_interval` in `handle_conf_change()` if we shouldn't dump the logs .